### ISSUE TITLE
Rehydration of Immutable structures causes a warning

### ISSUE THREAD

**Issue by jeremejevs** [rt2zz/redux-persist#57]
This is what I get when rehydrating an Immutable map:

```js
iterable.length has been deprecated, use iterable.size or iterable.count() *snip*
    at Map.Object.defineProperty.get
    at *snip* // in lodash.baseProperty
    at isArrayLike
    at isArrayLikeObject
    at isArguments
    at Object.isPlainObject [as default]
    at *snip* // in redux-persist.createRehydrationReducer
    at computeNextEntry
    at recomputeStates
    at *snip* // in redux.liftReducerWith
```

[The line of interest](https://github.com/rt2zz/redux-persist/blob/master/src/autoRehydrate.js#L53). It appears that `lodash.isplainobject` involves checking whether that object has a `length` property or not, and Immutable doesn't like that. There's a [very similar resolved issue](https://github.com/michaelcontento/redux-storage/issues/8) in `redux-storage`. [Here's their fix](https://github.com/michaelcontento/redux-storage/commit/1147c1296fb2d18aa41efb08eef802eae9e00c86#diff-46f56fee267eb512b656d27b24987144R19).

**Comment by jeremejevs** [comment]
[I see that there was a check, removed in 1.5.1](https://github.com/rt2zz/redux-persist/commit/7dabb03ade05dc6af2844d15424dbdccf09f1c94#diff-5b7c470ef7adce0d690c36d1d956ea6bL25).

**Comment by rt2zz** [comment]
Thanks for catching that. The old `size` check was ugly, I prefer redux-storage's method of checking for `mergeDeep`.

Updated in 1.5.5 :+1: : https://github.com/rt2zz/redux-persist/commit/dae5b1b58ae4b9b598e6bd5f421d136f1a0399a6

**Comment by jeremejevs** [comment]
Much appreciated :+1:

**Comment by namxam** [comment]
I get the same in 2.0.0. Are you planning a fix for that too?

**Comment by rt2zz** [comment]
@namxam my mistake, just released the analogous fix in 2.0: https://github.com/rt2zz/redux-persist/commit/8059972d5af168af4830c8b5c0c311ebf0573c24

I am not personally using immutable, so if you can verify it working that would be much appreciated!

**Comment by namxam** [comment]
Checked it and the warning is gone. Thanks!
